src/axom/quest/readers/ProEReader.cpp

```diff
@@ -83,21 +80,76 @@ int ProEReader::read()
   }
 
   // Initialize tets
+  int tet_count = 0;
   for(int i = 0; i < m_num_tets; i++)
   {
     ifs >> id >> tet_nodes[0] >> tet_nodes[1] >> tet_nodes[2] >> tet_nodes[3];
 
-    for(int j = 0; j < NUM_NODES_PER_TET; j++)
+    if(!m_tetPredicate || m_tetPredicate(tet_nodes, i, m_nodes))
     {
-      // Node IDs start at 1 instead of 0, adjust to 0 for indexing
-      m_tets.push_back(tet_nodes[j] - 1);
+      tet_count += 1;
+      for(int j = 0; j < NUM_NODES_PER_TET; j++)
+      {
+        // Node IDs start at 1 instead of 0, adjust to 0 for indexing
+        m_tets.push_back(tet_nodes[j] - 1);
+      }
     }
   }
 
   ifs.close();
+
+  compact_arrays(tet_count);
+
   return (0);
 }
 
+//------------------------------------------------------------------------------
+void ProEReader::setTetPredFromBoundingBox(BBox3D& box, bool inclusive)
+{
+  if(box.isValid())
+  {
+    if(!inclusive)
+    {
+      auto pred = [box](int tet_nodes[4],
+                        [[maybe_unused]] int id,
```

Comment:
I don't think we can use `[[maybe_unused]]` until we move to `c++17`. 
Please replace it w/ the `AXOM_UNUSED_PARAM` macro.
https://github.com/LLNL/axom/blob/0a47cb26b8dd5d358cdee6fa0653f326ff98f834/src/axom/core/Macros.hpp#L134-L146

See: https://en.cppreference.com/w/cpp/language/attributes/maybe_unused

Reply:
Very true.  I'll fix.

Reply:
Fixed by removing the name of the argument.